### PR TITLE
TextField added event return to onChange and updated the test

### PR DESCRIPTION
**Pull request by brilliant-ember**
### WHY are these changes introduced?
As part of fixing this https://github.com/Shopify/shipping/issues/3083 we need to pass the event to `addressLine` in web, and to that end the polaris TextField is being updated to allow passing the event. 
Next part of the above issue fix is to use the event to prevent the curser from jumping to the end.

Fixes #0000 

### WHAT is this pull request doing?
Allow TextField to pass the input event so that components that need it can access it.

## 

### How to 🎩
https://user-images.githubusercontent.com/23269759/123495307-394b4180-d5f1-11eb-9e61-eee7f5bdfecd.mp4

🖥 [Local development instructions]()
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Comment by alex-page** [comment]
@brilliant-ember we have intentionally not included the browser event api in this component. Is there an alternative approach you could try in your implementation? How are teams doing this in `shopify/web`?

**Comment by brilliant-ember** [comment]
> @brilliant-ember we have intentionally not included the browser event api. Is there an alternative approach you could try in your implementation? How are teams doing this in `shopify/web`?

other places don't use redux, this problem seems to happen when redux state store updates the textfield and there's a component-local state changing (loading spinner) The addressLine.tsx in web is used in multiple places but not all of them use redux, the shippingLabels page uses redux and that's where the problem is observed. So far I am talking with a senior FED who worked in that code part to see if we can somehow have a specific solution for the redux page, but if that's not possible I can't think of a solution other than somehow passing the event from polaris and programmatically place the curser using the selectedStart/End stored in the event.

**Comment by lucasfrosty** [comment]
Hi 👋 i'm the FED that @brilliant-ember mentioned.

Unfortunately using Redux for state management all the time (even for small fields inside modals) is a required thing on our application because of some reasons that are not worth going through right now. However we did some investigation today and find a workaround for this that won't require dealing with this cursor position hack... Basically we're gonna use local state, but that will be just a copy of the field contained in the Redux store, and we'll keep both in sync all the time, that way we'll update the store accordingly, and the cursor issue won't be a thing anymore since we'll be actually using local state instead.

So, to summarize, this PR will no longer be necessary.
